src/ChosenSubjects.js

```diff
@@ -1,24 +1,113 @@
 import React from 'react';
 import {makeStyles} from "@material-ui/core/styles";
 import {useSubjects} from "./hooks";
+import DeleteIcon from '@material-ui/icons/Delete';
+import { Button } from '@material-ui/core';
+import {DataGrid} from "@material-ui/data-grid";
+import {subjectColumns} from "./ChosenSubjectsColumns"
 
 const useStyles = makeStyles((theme) => ({
     root: {
         margin: "100px"
     }
 }));
 
+// vychozi razeni
+const sortModel = [
+    {
+        field: 'code',
+        sort: 'asc',
+    },
+];
+
+const getSubjectRows = subjectsFromJSON => {
+    // priprav predmety z JSONu pro zobrazeni v tabulce
+    // pro zobrazeni v tabulce je nutne, aby kazdy zaznam mel ciselne ID, to pak urcuje default sort
+    for (var i = 0; i < subjectsFromJSON.length; i++)
+    {
+        subjectsFromJSON[i].id = i;
+    }
+    return subjectsFromJSON;
+}
+
+const getSubjectColumns = subjects => {
+    // priprav sloupce dynamicky z dat
+    var columns = []
+    for (var prop in subjects[0])
+    {
+        if (prop != 'id')
+            columns.push(prop);
+    }
+    // nejak to serad
+    columns.sort();
+    // takhle to DataGrid vyzaduje - sloupce tabulky jsou samostatne objekty
+    columns = columns.map(x => {return {field: x.toString()}});
+    return columns;
+}
+
+
+const updateTotalRow = subjects => {
+    // prepocitej soucty v poslednim radku
+    let totalCredits = 0;
+    let totalLength = 0;
+    for (let i = 0; i < subjects.length; i++)
+    {
+        totalCredits += subjects[i].credits;
+        totalLength += Number(subjects[i]["length"]);
+    }
+
+    return [{id:0, credits:totalCredits, length:totalLength}]
```

Comment:
Tady bych pak v další iteraci vracel rovnou object {totalCredits: totalCredits, len: len}. Bacha na length, je to v js reserved.